test/parallel/test-http-pipeline-regr-3332.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+const net = require('net');
+
+const big = new Buffer(16 * 1024);
+big.fill('A');
+
+const COUNT = 1e4;
+
+var received = 0;
+
+var client;
+const server = http.createServer(function(req, res) {
+  res.end(big, function() {
+    if (++received === COUNT) {
+      server.close();
+      client.end();
+    }
+  });
+}).listen(common.PORT, function() {
+  var req = new Array(COUNT + 1).join('GET / HTTP/1.1\r\n\r\n');
+  client = net.connect(common.PORT, function() {
+    client.write(req);
+  });
```

Comment:
That allows data to be read, but not removed from the internal queue:
```
$ /usr/bin/time ./node test/parallel/test-http-pipeline-regr-3332.js 
0.43user 0.15system 0:00.54elapsed 107%CPU (0avgtext+0avgdata 123572maxresident)k
```
~123MB memory usage for this test.